tests/pytorch/test_datasets_pt.py

```diff
@@ -257,6 +207,313 @@ def test_svhn(input_size, size, rotate, mock_svhn_dataset):
+        [[512, 512], 3, False],
+    ],
+)
+def test_sroie(input_size, size, rotate, mock_sroie_dataset):
+    # monkeypatch the path to temporary dataset
+    datasets.SROIE.TRAIN = (mock_sroie_dataset, None, "sroie2019_train_task1.zip")
+
+    ds = datasets.SROIE(
+        train=True, download=True, sample_transforms=Resize(input_size), rotated_bbox=rotate,
+        cache_dir=mock_sroie_dataset, cache_subdir="sroie2019_train_task1",
+    )
+
+    assert len(ds) == size
+    assert repr(ds) == f"SROIE(train={True})"
+    img, target = ds[0]
+    assert isinstance(img, torch.Tensor)
+    assert img.shape == (3, *input_size)
+    assert img.dtype == torch.float32
+    assert isinstance(target, dict)
+    # depends on #702
+    assert isinstance(target['boxes'], np.ndarray)  # and np.all((target['boxes'] <= 1) & (target['boxes'] >= 0))
+    assert isinstance(target['labels'], list) and all(isinstance(s, str) for s in target['labels'])
+    assert len(target['labels']) == len(target['boxes'])
+
+    loader = DataLoader(
+        ds, batch_size=2, drop_last=True, sampler=RandomSampler(ds), num_workers=0, pin_memory=True,
+        collate_fn=ds.collate_fn)
+
+    images, targets = next(iter(loader))
+    assert isinstance(images, torch.Tensor) and images.shape == (2, 3, *input_size)
+    assert isinstance(targets, list) and all(isinstance(elt, dict) for elt in targets)
+
+
+@pytest.mark.parametrize(
+    "input_size, size, rotate",
+    [
+        [[512, 512], 3, True],  # Actual set has 149 training samples and 50 test samples
+        [[512, 512], 3, False],
+    ],
+)
+def test_funsd(input_size, size, rotate, mock_funsd_dataset):
+    # monkeypatch the path to temporary dataset
+    datasets.FUNSD.URL = mock_funsd_dataset
+    datasets.FUNSD.SHA256 = None
+    datasets.FUNSD.FILE_NAME = "funsd.zip"
+
+    ds = datasets.FUNSD(
+        train=True, download=True, sample_transforms=Resize(input_size), rotated_bbox=rotate,
+        cache_dir=mock_funsd_dataset, cache_subdir="funsd",
+    )
+
+    assert len(ds) == size
+    assert repr(ds) == f"FUNSD(train={True})"
+    img, target = ds[0]
+    assert isinstance(img, torch.Tensor)
+    assert img.shape == (3, *input_size)
+    assert img.dtype == torch.float32
+    assert isinstance(target, dict)
+    # depends on #702
+    assert isinstance(target['boxes'], np.ndarray)  # and np.all((target['boxes'] <= 1) & (target['boxes'] >= 0))
+    assert isinstance(target['labels'], tuple) and all(isinstance(s, str) for s in target['labels'])
+    assert len(target['labels']) == len(target['boxes'])
+
+    loader = DataLoader(
+        ds, batch_size=2, drop_last=True, sampler=RandomSampler(ds), num_workers=0, pin_memory=True,
+        collate_fn=ds.collate_fn)
+
+    images, targets = next(iter(loader))
+    assert isinstance(images, torch.Tensor) and images.shape == (2, 3, *input_size)
+    assert isinstance(targets, list) and all(isinstance(elt, dict) for elt in targets)
```

Comment:
We could define a function `_test_dataset_pt` which does all those steps, so that we can reduce the code duplication for all of those :+1: